src/v2/view-builder/internals/BaseForm.js

```diff
@@ -130,20 +130,30 @@ export default Form.extend({
   },
 
   /*
-   * Renders the contents of messages object (if any) during initialize
+   * Renders the contents of messages object (if any, on error) during initialize
    * This function is called during Form.initialize, and will display
    * messages when the form renders.
-   * Note: any errors happening after the form is rendered won't use this function,
-   * this is only used during initialize
+   * Note: Any user action related form errors handled by FormController
    */
-  showMessages() {
-    // render messages as text
-    const messagesObjs = this.options.appState.get('messages');
-    if (messagesObjs?.value.length) {
-      const content = messagesObjs.value.map((messagesObj) => {
-        return messagesObj.message;
+  showMessages(options) {
+    const messages = this.options.appState.get('messages') || {};
+    if (Array.isArray(messages.value)) {
+      messages.value.forEach(obj => {
+        if(!obj?.class || obj.class === 'INFO') {
+          this.add(`<div class="ion-messages-container">${obj.message}</div>`, '.o-form-error-container');
+        } else {
+          if(options instanceof View) {
+            this.add(options, '.o-form-error-container');
+          } else {
+            obj = Object.assign(obj, options);
```

Comment:
Why do you want to add options to the obj?

Reply:
consuming views can add callout options like `title`.. few views did that.

Reply:
i will rather do this.. thanks for pointing that out.
```
            options = Object.assign(options, _.pick(obj, 'message', 'class'));
```